src/utils/model_utils.py

```diff
@@ -169,7 +175,7 @@ def test(self, model, dloader, loss_fn, device, **kwargs) -> Tuple[float, float]
 
     def save_model(self, model, path):
         if type(model) == DataParallel:
-            model_ = model.module
+            model_ = model
```

Comment:
this is incorrect in my opinion. DataParallel instance has to be saved with `.module` param